pkg/controllers/job/plugins/distributed-framework/mpi/mpi.go

```diff
@@ -82,7 +82,10 @@ func (mp *Plugin) OnPodCreate(pod *v1.Pod, job *batch.Job) error {
 	workerHosts := ""
 	env := v1.EnvVar{}
 	if helpers.GetTaskKey(pod) == mp.masterName {
-		workerHosts = mp.generateTaskHosts(job.Spec.Tasks[helpers.GetTaskIndexUnderJob(mp.workerName, job)], job.Name)
+		taskIndex := helpers.GetTaskIndexUnderJob(mp.workerName, job)
+		if taskIndex >= 0 {
```

Comment:
When there is no worker, the env value of MPIHOST will be empty, will this cause an error, shouldn't we set this env?